### PR TITLE
updated dropsonde code to latest JOANNE version

### PR DESCRIPTION
**Pull request by d70-t**
The JOANNE dataset received an update which changed the sonde dimension from `sounding` to `sonde_id` and removed the `PW` variable. This made several changes throughout the book necessary.

### CODE REVIEW

**Comment by d70-t** [comment]
The build fails because `sonde_id` and `platform_id` are being read as bytrstring objects when opened via OPeNDAP, which would require `.decode("utf-8")`. However, when the datasets are opened via zarr (as it was the case before), the string are read in as already decoded strings.
Doing the decoding manually creates a bit of a mess, especially when done conditionally. That's why I'm still hesitant to to the "simple fix".

**Comment by d70-t** [comment]
Due to the merge of eurec4a/eurec4a-intake#80, the updated dropsonde code now does work.
The checks should pass after eurec4a/eurec4a-intake#81 is merged.

File: how_to_eurec4a/dropsondes.md [hunk 1]
```diff
@@ -138,40 +138,30 @@ ax0.legend([dt64_to_dt(d).strftime("%H:%M:%S")
...
-    xticks = np.arange(0, ds_sondes_Feb05.sounding.size, int(ds_sondes_Feb05.sounding.size/10))
-    ax.set_xticks(xticks)
-    ax.set_xticklabels([dt64_to_dt(t).strftime("%H:%M") for t in ds.launch_time.values[xticks]])
-    ax.set_xlabel(f"{ds_sondes_Feb05.launch_time.long_name}")
-    yticks = np.arange(0, ds_sondes_Feb05.alt.size, int(ds_sondes_Feb05.alt.size/5))
-    ax.set_yticks(yticks)
-    ax.set_yticklabels(ds_sondes_Feb05.alt.values[yticks])
-    ax.set_ylabel(f"{ds_sondes_Feb05.alt.name} / m")
+    ds_sondes_Feb05.wspd.load().plot(ax=ax, x="launch_time", y="alt")
+    ax.xaxis.set_major_formatter(mdates.DateFormatter("%H:%M"))
```
Comment (tmieslinger) [inline how_to_eurec4a/dropsondes.md:165]: Ups and thanks! The old workaround proved to be non-sustainable when adding P3 data :D using the matplotlib.dates is a nice way.

One can see that the interpolation smooths the time steps when there is no sonde launched. Can you think of a way to better indicate (probably by including NaNs) that there is no data. In principle, I also like that the width of the bar indicates the sampling frequency. Maybe it's fair to note that somewhere in the text, too?

File: how_to_eurec4a/specmacs.md [hunk 2]
```diff
@@ -75,7 +75,7 @@ cat = eurec4a.get_intake_catalog()
 ```{code-cell} ipython3
 dropsondes = cat.dropsondes.JOANNE.level3.to_dask()
 # this following line should go away in the soon to be released new version of JOANNE
-sonde_dt = dropsondes.swap_dims({"sounding": "sonde_id"}).sel(sonde_id=first_dropsonde).launch_time.values
+sonde_dt = dropsondes.sel(sonde_id=first_dropsonde).launch_time.values
```
Comment (tmieslinger) [inline how_to_eurec4a/specmacs.md:78]: is the comment in line 77 still appropriate?

**Review by tmieslinger — APPROVED** [review]
Overall some really nice and necessary adjustments. Thanks for not only updating but also optimizing the code :)
I only have few minor suggestions.

File: how_to_eurec4a/dropsondes.md (hunk 1 above)
Comment (d70-t) [reply how_to_eurec4a/dropsondes.md]: I've added nearest neighbor interpolation with maximum distance limit. This shows white spaces for larger measurement gaps but closes the gaps when sondes are released quickly after each other.

File: how_to_eurec4a/specmacs.md (hunk 2 above)
Comment (d70-t) [reply how_to_eurec4a/specmacs.md]: Good catch! I removed it.

**Pull request merged**